common/client/node_fsm.go

```diff
@@ -0,0 +1,266 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+var (
+	promPoolRPCNodeTransitionsToAlive = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_alive",
+		Help: transitionString(NodeStateAlive),
+	}, []string{"chainID", "nodeName"})
+	promPoolRPCNodeTransitionsToInSync = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_in_sync",
+		Help: fmt.Sprintf("%s to %s", transitionString(NodeStateOutOfSync), NodeStateAlive),
+	}, []string{"chainID", "nodeName"})
+	promPoolRPCNodeTransitionsToOutOfSync = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_out_of_sync",
+		Help: transitionString(NodeStateOutOfSync),
+	}, []string{"chainID", "nodeName"})
+	promPoolRPCNodeTransitionsToUnreachable = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_unreachable",
+		Help: transitionString(NodeStateUnreachable),
+	}, []string{"chainID", "nodeName"})
+	promPoolRPCNodeTransitionsToInvalidChainID = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_invalid_chain_id",
+		Help: transitionString(NodeStateInvalidChainID),
+	}, []string{"chainID", "nodeName"})
+	promPoolRPCNodeTransitionsToUnusable = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "pool_rpc_node_num_transitions_to_unusable",
+		Help: transitionString(NodeStateUnusable),
+	}, []string{"chainID", "nodeName"})
+)
+
+// NodeState represents the current state of the node
+// Node is a FSM (finite state machine)
+type NodeState int
```

Comment:
Can this type and the consts be unexported?
```suggestion
type nodeState int
```

Reply:
NodeState was used in Node mock, but it turns out Node mode can be removed, so I unexported the state.